### PR TITLE
Billing apis

### PR DESCRIPTION
**Pull request by sangmin**
Billing update for viewing usage reports (EUCA-12976)

### CODE REVIEW

File: src/main/java/com/github/sjones4/youcan/youbill/YouEc2Reports.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import com.amazonaws.AmazonClientException;
+import com.github.sjones4.youcan.youbill.model.ViewInstanceUsageReportRequest;
+import com.github.sjones4.youcan.youbill.model.ViewInstanceUsageReportResult;
+import com.github.sjones4.youcan.youbill.model.ViewReservedInstanceUtilizationReportRequest;
+import com.github.sjones4.youcan.youbill.model.ViewReservedInstanceUtilizationReportResult;
+
+/*
+ *
+ */
+public interface YouEc2Reports {
```
Comment (sjones4) [inline src/main/java/com/github/sjones4/youcan/youbill/YouEc2Reports.java]: This is a separate service so should be in its own package.

File: src/main/java/com/github/sjones4/youcan/youbill/model/ViewInstanceUsageReportResult.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+  @Override
+  public boolean equals( Object obj ) {
+    if ( this == obj ) return true;
+    if ( obj == null ) return false;
+
+    if ( obj instanceof ViewInstanceUsageReportResult == false ) return false;
+    final ViewInstanceUsageReportResult other = (ViewInstanceUsageReportResult) obj;
+
+    if (other.getUsageReport() == null ^ this.getUsageReport() == null) return false;
```
Comment (sjones4) [inline src/main/java/com/github/sjones4/youcan/youbill/model/ViewInstanceUsageReportResult.java]: Should also check value when not null

File: src/main/java/com/github/sjones4/youcan/youbill/model/ViewMonthlyUsageRequest.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  @Override
+  public boolean equals(Object obj) {
+    if (this == obj) return true;
+    if (obj == null) return false;
+
+    if (obj instanceof ViewMonthlyUsageRequest == false) return false;
+    final ViewMonthlyUsageRequest other = (ViewMonthlyUsageRequest)obj;
+
+    if (other.getYear() == null ^ this.getYear() == null) return false;
+    if (other.getMonth() == null ^ this.getMonth() == null) return false;
```
Comment (sjones4) [inline src/main/java/com/github/sjones4/youcan/youbill/model/ViewMonthlyUsageRequest.java]: Should also check values when not null

File: src/main/java/com/github/sjones4/youcan/youbill/model/ViewMonthlyUsageResult.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+  @Override
+  public boolean equals( Object obj ) {
+    if ( this == obj ) return true;
+    if ( obj == null ) return false;
+
+    if ( obj instanceof ViewUsageResult == false ) return false;
+    final ViewUsageResult other = (ViewUsageResult) obj;
+
+    if (other.getData() == null ^ this.getData() == null) return false;
```
Comment (sjones4) [inline src/main/java/com/github/sjones4/youcan/youbill/model/ViewMonthlyUsageResult.java]: Should also check value when not null

File: src/main/java/com/github/sjones4/youcan/youbill/model/ViewReservedInstanceUtilizationReportResult.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+  @Override
+  public boolean equals( Object obj ) {
+    if ( this == obj ) return true;
+    if ( obj == null ) return false;
+
+    if ( obj instanceof ViewReservedInstanceUtilizationReportResult == false ) return false;
+    final ViewReservedInstanceUtilizationReportResult other = (ViewReservedInstanceUtilizationReportResult) obj;
+
+    if (other.getUtilizationReport() == null ^ this.getUtilizationReport() == null) return false;
```
Comment (sjones4) [inline src/main/java/com/github/sjones4/youcan/youbill/model/ViewReservedInstanceUtilizationReportResult.java]: Should also check value when not null

File: src/main/java/com/github/sjones4/youcan/youbill/model/ViewUsageRequest.java [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+    if (obj instanceof ViewUsageRequest == false) return false;
+    ViewUsageRequest other = (ViewUsageRequest)obj;
+
+    if (other.getServices() == null ^ this.getServices() == null) return false;
+    if (other.getUsageTypes() == null ^ this.getUsageTypes() == null) return false;
+    if (other.getOperations() == null ^ this.getOperations() == null) return false;
+    if (other.getTimePeriodFrom() == null ^ this.getTimePeriodFrom() == null) return false;
+    if (other.getTimePeriodTo() == null ^ this.getTimePeriodTo() == null) return false;
+    if (other.getReportGranularity() == null ^ this.getReportGranularity() == null) return false;
```
Comment (sjones4) [inline src/main/java/com/github/sjones4/youcan/youbill/model/ViewUsageRequest.java]: Should also check values when not null

File: src/main/java/com/github/sjones4/youcan/youbill/model/ViewUsageResult.java [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+  @Override
+  public boolean equals( Object obj ) {
+    if ( this == obj ) return true;
+    if ( obj == null ) return false;
+
+    if ( obj instanceof ViewUsageResult == false ) return false;
+    final ViewUsageResult other = (ViewUsageResult) obj;
+
+    if (other.getData() == null ^ this.getData() == null) return false;
```
Comment (sjones4) [inline src/main/java/com/github/sjones4/youcan/youbill/model/ViewUsageResult.java]: Should also check value when not null

**Review by sjones4 — CHANGES_REQUESTED** [review]
Looks good to me other than:
* ec2reports service should be in its own package. 
* equality checks for model classes are missing value checks

**Comment by sangmin** [comment]
@sjones4 change has been committed reflecting your review.

**Pull request merged**